gobblin-temporal/src/main/java/org/apache/gobblin/temporal/ddm/workflow/impl/ProcessWorkUnitsWorkflowImpl.java

```diff
@@ -72,20 +72,33 @@ private CommitStats performWork(WUProcessingSpec workSpec) {
     searchAttributes = TemporalWorkFlowUtils.generateGaasSearchAttributes(jobState.getProperties());
 
     NestingExecWorkflow<WorkUnitClaimCheck> processingWorkflow = createProcessingWorkflow(workSpec, searchAttributes);
-    int workunitsProcessed =
-        processingWorkflow.performWorkload(WorkflowAddr.ROOT, workload, 0, workSpec.getTuning().getMaxBranchesPerTree(),
-            workSpec.getTuning().getMaxSubTreesPerTree(), Optional.empty());
-    if (workunitsProcessed > 0) {
-      CommitStepWorkflow commitWorkflow = createCommitStepWorkflow(searchAttributes);
-      CommitStats result = commitWorkflow.commit(workSpec);
-      if (result.getNumCommittedWorkUnits() == 0) {
-        log.warn("No work units committed at the job level. They could have been committed at the task level.");
-      }
-      return result;
-    } else {
+
+    Optional<Integer> workunitsProcessed = Optional.empty();
+    try {
+      workunitsProcessed = Optional.of(processingWorkflow.performWorkload(WorkflowAddr.ROOT, workload, 0,
+          workSpec.getTuning().getMaxBranchesPerTree(), workSpec.getTuning().getMaxSubTreesPerTree(),
+          Optional.empty()));
+    } catch (Exception e) {
+      log.error("ProcessWorkUnits failure - will attempt partial commit before announcing error", e);
+      performCommitIfAnyWorkUnitsProcessed(workSpec, searchAttributes, workunitsProcessed);
+      throw e; //We want to proceed with partial commit and throw exception so that the parent workflow ExecuteGobblinWorkflowImpl can throw the failure event
```

Comment:
need space after `//` and also could streamline comment:
```
throw e; // re-throw after any partial commit, to fail the parent workflow
```

but anyway, that `throw` on L84 won't run will it?  the partial commit (`CommitStepWorkflow`) will throw an exception that unwinds the stack beforehand.  am I correct that at best this `throw` is a "should never happen" fallback?

ITO which we'd prefer to surface (possibly both) - what specific info is in the orig exception relative to the one currently being throw?  would it make sense to combine messages from the two or simply throw one or the other?

Reply:
addressed, caught exception from commit step and threw exception with both the messages combined